src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -0,0 +1,1956 @@
+                passed to generate matching the attributes of `generation_config` will override them. If
+                `generation_config` is not provided, the default will be used, which had the following loading
+                priority: 1) from the `generation_config.json` model file, if it exists; 2) from the model
+                configuration. Please note that unspecified parameters will inherit [`~generation.GenerationConfig`]'s
+                default values, whose documentation should be checked to parameterize generation.
+
+        Returns:
+            `ClvpOutput` or tuple: A `ClvpOutput` (if `return_dict_in_generate=True` or when
+            `config.return_dict_in_generate=True`) or a tuple.
+        """
+
+        if generation_config is None:
+            generation_config = self.generation_config
+        generation_config.update(**kwargs)
+
+        conditioning_embeds = self.conditioning_encoder(mel_spec=input_features, input_ids=input_ids)
+
+        speech_candidates = self.speech_decoder_model.generate(
+            conditioning_embeds=conditioning_embeds,
+            generation_config=generation_config,
+        )
+        if isinstance(speech_candidates, ModelOutput):
+            speech_candidates = speech_candidates.sequences
+        speech_candidates = self.fix_speech_decoder_output(speech_candidates)
+
+        speech_outputs = self.speech_encoder_model(
+            input_ids=speech_candidates,
+        )
+        text_outputs = self.text_encoder_model(
+            input_ids=input_ids,
+        )
+
+        speech_embeds = speech_outputs[0]
+        text_embeds = text_outputs[0]
+
+        # normalized features
+        speech_embeds = speech_embeds / speech_embeds.norm(p=2, dim=-1, keepdim=True)
+        text_embeds = text_embeds / text_embeds.norm(p=2, dim=-1, keepdim=True)
+
+        # cosine similarity as logits
+        logit_scale = self.logit_scale.exp()
+        logits_per_text = torch.matmul(text_embeds, speech_embeds.t()) * logit_scale
+        logits_per_speech = logits_per_text.t()
+
+        if not generation_config.return_dict_in_generate:
+            output = (
+                speech_candidates,
+                logits_per_speech,
+                logits_per_text,
+                text_embeds,
+                speech_embeds,
+                text_outputs[2],
+                speech_outputs[2],
+            )
+            return output
+
+        return ClvpOutput(
+            speech_candidates=speech_candidates,
+            logits_per_speech=logits_per_speech,
+            logits_per_text=logits_per_text,
+            text_embeds=text_embeds,
+            speech_embeds=speech_embeds,
+            text_model_output=text_outputs[2],
+            speech_model_output=speech_outputs[2],
+        )
+
+    def prepare_inputs_for_generation(self, input_ids, input_features, **kwargs):
+        return {"input_ids": input_ids, "input_features": input_features, **kwargs}
```

Comment:
No past-key values? This is going to make auto-regressive decoding very slow. Maybe you can take inspiration from Whisper here to speed it up: https://github.com/huggingface/transformers/blob/1a2e966cfe916cdfd78ef7e1385911106bc33f50/src/transformers/models/whisper/modeling_whisper.py#L1787

Reply:
lets remove it.